RPIPs/RPIP-72.md

```diff
@@ -0,0 +1,164 @@
+| `notify_threshold`              | uint256 | `12 hours`    | \≥ 2 hours         |
+| `late_notify_fine`              | uint256 | `0.05 ETH`    | ≤ 0.5 ETH          |
+| `user.distribute.window.length` | uint256 | `7 days`      | \≥ 1 day; ≤30 days |
+```
+
+The following section of RPIP-43
+```md
+- There SHALL be a capital distribution function in the megapool
+  - When called, capital borrowed from the protocol that has been released from
+    exited validators SHALL be sent to the rETH contract.
+    - If the capital is insufficient to repay the protocol, the shortfall SHALL be added to `debt`
+  - When called while the megapool has `debt`, the remaining capital from exited validators SHALL first be used to pay off `debt`
+  - When called, the remaining capital SHALL then be held in the megapool as unclaimed node operator funds
+  - This function SHALL support permissionless use, with additional restrictions:
+    - A permissionless `startUserDistribute` function SHALL record the time when called, as long as there is no previously recorded time _or_ the previously recorded time was longer than `minipool.user.distribute.window.start + minipool.user.distribute.window.length` ago
+    - The capital distribution function may be called permissionlessly if `minipool.user.distribute.window.start` has passed since the recorded `startUserDistribute` call, but `minipool.user.distribute.window.start + minipool.user.distribute.window.length` has not yet passed since the recorded `startUserDistribute` call
+```
+SHALL be replaced by
+```md
+- There SHALL be a capital distribution process in the megapool
+  - 
+  - There SHALL be a permissionless `notify_exit` function, which proves the `withdrawable_epoch` and is intended to be called by a node operator per validator they exit
+    - On notification, distribution of rewards SHALL be prevented after the earliest validator `withdrawal_epoch`, until all exiting validators have had `notify_final_balance` called
+  - Any oDAO member SHALL be able to create an exit challenge if a node operator does not call `notify_exit` more than `notify_threshold` before `withdrawable_epoch`
+    - There MAY be additional restrictions to which oDAO members can challenge (eg, not the oDAO member that most recently challenged)
+    - There MAY be support for batching multiple exit challenges
+    - Until an exit challenge is resolved, a node operator SHALL NOT be able to distribute rewards 
+    - An exit challenge SHALL be resolved when either:
+      - `notify_exit` is called successfully, in which case the validator is considered exiting; `debt` SHALL be increased by `late_notify_fine`
+      - a proof that the validator in question is _not_ exiting is provided (eg, showing that `withdrawable_epoch` is `FAR_FUTURE_EPOCH`). In this case, the validator will not be considered exiting.
+  - There SHALL be a `notify_final_balance` function, which can be called once exited validators are withdrawable. This function MUST:
```

Comment:
Possibly be explicit about whether this can be batched?

Reply:
sure -- moved the batching comment lower out a level of hierarchy and noted all 3 functions as MAY be batched